core/doc.txt

```diff
@@ -25,3 +25,9 @@
  * @ingroup     core
  * @brief       Configuration data and startup code for the kernel
  */
+
+ /**
+  * @defgroup    core_sync Synchronization
```

Comment:
Maybe call the group `Thread Synchronization` ?

Reply:
sounds good